### ISSUE TITLE
Troubles parsing a C++ project with Clang.jl v0.18.0 and julia v1.11 (llvm 16)

### ISSUE THREAD

**Issue by giordano** [JuliaInterop/Clang.jl#482]
I'm having troubles parsing a C++ project with Clang.jl v0.18.0 and julia v1.11 (which uses llvm16).  To be clear, my current parsing script works fine with same version of Clang.jl but using julia v1.7-v1.10 (llvm from 12 to 15).  This may be related to #470.

My parsing script is at https://github.com/JuliaIPU/IPUToolkit.jl/blob/dca1f4d514a4f8184967da10b5b2b7e6919d8789/deps/generate_wrapper.jl (@Gnimuc contributed to it before :slightly_smiling_face:), getting the files to parse is a bit complicated, but there's a docker container you can use ([`graphcore/poplar:3.3.0-ubuntu-20.04-20230703`](https://hub.docker.com/layers/graphcore/poplar/3.3.0-ubuntu-20.04-20230703/images/sha256-53d5a4ea2d7e3c8b31468181a6947773d34f3856d7cf28bbe10cf83ccdf39f08?context=explore)), but it's possible the parsing script just has to be updated (although, again, it's working fine with other versions of julia/llvm).

For reference, I'm attaching below the diff of a couple of files generated by the parsing script
[gen_inherit.cpp.patch.txt](https://github.com/JuliaInterop/Clang.jl/files/14918987/gen_inherit.cpp.patch.txt)
[gen_inline.cpp.patch.txt](https://github.com/JuliaInterop/Clang.jl/files/14918988/gen_inline.cpp.patch.txt)

**Comment by Gnimuc** [comment]
Julia 1.11 uses llvm16 and there are some breaking changes introduced in llvm16/clang16(see #465). 

As IPUToolkit.jl maintains its own C++ parsing script, we should upgrade the script to be compatible with clang16.

**Comment by giordano** [comment]
I had a look at the [0.18 changelog](https://juliainterop.github.io/Clang.jl/v0.18.0/generated_changelog/#[v0.18.0](https://github.com/JuliaInterop/Clang.jl/releases/tag/v0.18.0)-2024-04-08) (I was very glad to see it!) but the only breaking change mentioned didn't look relevant, that's why I opened the issue :slightly_smiling_face:

**Comment by Gnimuc** [comment]
You're right. It's safe until Julia v1.1 is released. But if you're only using Clang.jl for generating bindings, it should be ok to remove it as a dependency.

**Comment by giordano** [comment]
> But if you're only using Clang.jl for generating bindings, it should be ok to remove it as a dependency.

I wish I could do that, bu the problem is that I have no other way to generate the bindings to start with :slightly_smiling_face:

**Comment by giordano** [comment]
I presume there hasn't been any progress with regards to C++ support for LLVM 16+, right?
